src/app/services/schedule.service.ts

```diff
@@ -13,13 +14,47 @@ export class ScheduleService {
   /**
    * Method gets schedule for the class by id
    * @param classId - Id of class
-   * @returns - Array with schedule data
+   * @returns - Array with schedule data or error if that happens
    */
-  getSchedule(classId: number): Observable<ScheduleData> {
+  getSchedule(classId: number): Observable<any> {
     return this.http.get(`/classes/${classId}/schedule`).pipe(map(
-      (response: {status: any, data: ScheduleData}) => {
-        return response.data;
-      }
-    ));
+      (response: {status: any, data: ScheduleData}) => response.data
+    ),
+    catchError(error => {
+      console.log(error);
+      return throwError('Розклад для вибраного класу не отримано. Спробуйте ще раз');
```

Comment:
do we really need to throw such errors? I believe error messages should be handled in component